### PR TITLE
#177182814 Replace AUTH0 token with JWT

### PR DESCRIPTION
**Pull request by barbarawuk**
These need adding to your .env first:
JWT_ALGORITHM=HS256
JWT_SECRET=YOU_MUST_CHANGE_THIS_PASSWORD

Admins can create new api users:

add api user

Then generate a token for them:
generate_new_token

after_token_generation

Tokens currently have no expiry but can be revoked if needs be

Still to do - add authorization so only admins can create api users (I'll add this as a separate PT story as needs doing for all admin pages)

### CODE REVIEW

File: app/controllers/admin/api_users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  end
+
+  def show; end
+
+  def generate_token
+    timestamp = Time.current
+    token = generate_api_user_token(timestamp)
+    if token && @api_user.update({ token_sample: token.last(5),
+                                   token_generated_at: timestamp,
+                                   token_generated_by: current_user.name })
```
Comment (artur79) [inline app/controllers/admin/api_users_controller.rb:30]: why just not user id and than read user name ?

File: spec/features/admin/api_users/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  let(:contact_email) { Faker::Internet.email }
+
+  before do
+    log_in admin
+    visit admin_api_users_path
+    click_link I18n.t('actions.add', model_name: ApiUser.model_name.human)
+  end
+
+  it 'creates API user' do
+    fill_in ApiUser.human_attribute_name('name'), with: name
```
Comment (artur79) [inline spec/features/admin/api_users/create_spec.rb:16]: I saw that pattern in many specs in Murfin, but I'm still not convinced to that way of testing forms. Test should say verbosely what label exactly we expect imho.

File: spec/features/admin/api_users/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    fill_in ApiUser.human_attribute_name('contact_email'), with: contact_email
+    click_button I18n.t('actions.save')
```
Comment (artur79) [inline spec/features/admin/api_users/create_spec.rb:18]: ... same here

File: spec/features/admin/api_users/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    expect(page).to have_content(I18n.t('notice.successfully.created', model_name: ApiUser.model_name.human))
```
Comment (artur79) [inline spec/features/admin/api_users/create_spec.rb:20]: here maybe it's justified cuz it's kinda generic message, but not so readable as just expected string.

**Review by artur79 — APPROVED** [review]

File: app/controllers/admin/api_users_controller.rb (hunk 1 above)
Comment (barbarawuk) [reply app/controllers/admin/api_users_controller.rb:30]: Primarily this is in case the user that generated the token is subsequently deleted.

Comment (artur79) [reply app/controllers/admin/api_users_controller.rb:30]: makes sense

File: spec/features/admin/api_users/create_spec.rb (hunk 2 above)
Comment (barbarawuk) [reply spec/features/admin/api_users/create_spec.rb:16]: Yeah, I'm not convinced either tbh. I was following the convention that was already in existing specs.
I guess it's quite likely that on other installations the contents of the en_GB.yml would be changed and that would then break the tests. The question is would that matter, is anyone else likely to be running the tests or be contributors... I'm not sure?
I'll put it up as a new story on PT and we can discuss in tech review.

Comment (artur79) [reply spec/features/admin/api_users/create_spec.rb:16]: good to know that I'm not alone with this preference :)

**Pull request merged**